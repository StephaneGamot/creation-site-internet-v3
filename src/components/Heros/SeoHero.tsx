"use client";

import { motion, useReducedMotion } from "framer-motion";
import Link from "next/link";
import { useTranslations, useLocale } from "next-intl";

export default function SeoHero(){
  const t = useTranslations("seoHero");
  const locale = useLocale();
  const shouldReduceMotion = useReducedMotion();

  return (
    <section
      aria-labelledby="seo-hero-title"
      className="text-center pt-12 max-w-4xl mx-auto mb-12"
    >
      <motion.h1
        id="seo-hero-title"
        initial={{ opacity: 0, y: 40 }}
        animate={shouldReduceMotion ? { opacity: 1 } : { opacity: 1, y: 0 }}
        transition={{ duration: 0.8 }}
        className="text-3xl md:text-4xl font-title font-semibold mb-6"
      >
        {t("title")}
      </motion.h1>
      <p className="text-lg md:text-xl text-gray-700 mb-8">{t("text")}</p>
      <Link
        href={`/${locale}/stephane-gamot`}
        className="inline-block bg-[#43986b] text-white px-6 py-3 rounded-xl font-semibold shadow hover:bg-opacity-90 transition"
        aria-label={t("aria")}
      >
        {t("button")}
      </Link>
    </section>
  );
}
