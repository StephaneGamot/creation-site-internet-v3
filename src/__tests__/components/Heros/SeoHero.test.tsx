import React from "react";
import { render, screen } from "@testing-library/react";
import SeoHero from "@/components/Heros/SeoHero";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

// ✅ Mock des traductions
jest.mock("next-intl", () => ({
  useTranslations: () => {
    const t = (key: string) => {
      const translations: Record<string, string> = {
        title: "Boostez votre SEO avec un site performant",
        text: "Un bon référencement commence dès la conception.",
        button: "En savoir plus",
        aria: "Accéder à la section Stéphane Gamot",
      };
      return translations[key] || "";
    };
    return t;
  },
  useLocale: () => "fr",
}));

// ✅ Mock de framer-motion
jest.mock("framer-motion", () => ({
  motion: {
    h1: (props: any) => <h1 {...props} />,
  },
  useReducedMotion: () => false,
}));

describe("SeoHero", () => {
  it("affiche le titre, le texte et le bouton", () => {
    render(<SeoHero />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
      "Boostez votre SEO avec un site performant"
    );
    expect(
      screen.getByText("Un bon référencement commence dès la conception.")
    ).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveTextContent("En savoir plus");
    expect(screen.getByLabelText("Accéder à la section Stéphane Gamot")).toBeInTheDocument();
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<SeoHero />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const tree = renderer.create(<SeoHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
