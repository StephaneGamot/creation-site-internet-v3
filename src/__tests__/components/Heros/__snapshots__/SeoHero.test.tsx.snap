// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SeoHero correspond au snapshot 1`] = `null`;
